jerry-main/main-unix-snapshot.c

```diff
@@ -391,13 +391,15 @@ process_generate (cli_state_t *cli_state_p, /**< cli state */
 
   if (literals_file_name_p != NULL && !is_import_literals)
   {
-    const size_t literal_buffer_size = jerry_parse_and_save_literals ((jerry_char_t *) source_p,
-                                                                      source_length,
-                                                                      false,
-                                                                      output_buffer,
-                                                                      sizeof (output_buffer) / sizeof (uint32_t),
-                                                                      is_save_literals_mode_in_c_format);
-    if (literal_buffer_size == 0)
+    uint32_t *lit_out_buf_p = output_buffer + snapshot_size;
```

Comment:
`output_buffer` is an `uint32_t` array, `snapshot_size` seems to be a expressed in bytes. Pointer arithmetics will cause the addition to be off by a factor of 4 than expected.